hotham/src/engine.rs

```diff
@@ -25,34 +25,40 @@ pub static ANDROID_LOOPER_NONBLOCKING_TIMEOUT: Duration = Duration::from_millis(
 #[cfg(target_os = "android")]
 pub static ANDROID_LOOPER_BLOCKING_TIMEOUT: Duration = Duration::from_millis(i32::MAX as _);
 
-/// The Hotham Engine
-/// A wrapper around the "external world" from the perspective of the engine, eg. renderer, XR, etc.
-/// **IMPORTANT**: make sure you call `update` each tick
-pub struct Engine {
-    should_quit: Arc<AtomicBool>,
-    #[allow(dead_code)]
-    resumed: bool,
-    event_data_buffer: EventDataBuffer,
-    /// OpenXR context
-    pub xr_context: XrContext,
-    /// Vulkan context
-    pub vulkan_context: VulkanContext,
-    /// Renderer context
-    pub render_context: RenderContext,
-    /// Physics context
-    pub physics_context: PhysicsContext,
-    /// Audio context
-    pub audio_context: AudioContext,
-    /// GUI context
-    pub gui_context: GuiContext,
-    /// Haptics context
-    pub haptic_context: HapticContext,
+/// Builder for `Engine`.
+#[derive(Default)]
+pub struct EngineBuilder<'a> {
+    application_name: Option<&'a str>,
+    application_version: Option<u32>,
+    openxr_extensions: Option<xr::ExtensionSet>,
 }
 
-impl Engine {
-    /// Create a new instance of the engine
-    /// NOTE: only one instance may be running at any one time
+impl<'a> EngineBuilder<'a> {
+    /// Create an `EngineBuilder`
     pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Set the OpenXR application name
+    pub fn application_name(&mut self, name: Option<&'a str>) -> &mut Self {
```

Comment:
Interesting - a builder pattern is definitely where I wanted to end up but I've been a bit shy about creating one. But this does feel like a reasonable start since up until this point we've not really allowed any customisation.

Vulkan also has an `application_name` field (or similar). Think you could plumb this down into `VulkanContext` as well?